firefox-ios/Client/Frontend/Browser/Tabs/Views/TabDisplayView.swift

```diff
@@ -102,12 +102,24 @@ class TabDisplayView: UIView,
     }
 
     func newState(state: TabsPanelState) {
+        // We only want to respond to `TabsPanelState` for the current `panelType` (e.g. don't scroll the regular tabs for
+        // a private tabs scroll)
+        switch panelType {
+        case .tabs:
+            guard !state.isPrivateMode else { return }
+        case .privateTabs:
+            guard state.isPrivateMode else { return }
+        case .syncedTabs:
+            // FIXME what about the synced tab panel?
+            break
+        }
+
```

Comment:
@OrlaM I didn't realize until you told me that there can be multiple `TabDisplayView`s each waiting for `TabsPanelState` changes, and therefore we should be checking the `panelType` each time.  

I think the problem I have with this current state structure is, if the whole app had to render based on its global state, we are missing state for one of the `TabDisplayView`s. It literally doesn't exist in the global state because `TabsPanelState` only contains state for a single panel type at any given time. 🤔  (I realize the state exists somewhere in a different form since we're switching between panels, but from the perspective of a single `TabDisplayView` it doesn't).

Do you have any concerns about this as well, or do you think this is acceptable practice for our redux implementation?

I tried making a diagram to better illustrate my concern:
![TabPanelState drawio](https://github.com/user-attachments/assets/ef487f5f-c9e4-424b-94fb-cca6e48f1b22)

Reply:
Ahhh you're right! 🤔  Thank you for clarifying, I misunderstood the underlying representation.

Ok, so where I think I'm getting caught up is with the fact that ANY `TabsPanelState` update triggers the `newState()` call for ALL the `TabDisplayPanel`s at once... When I worked with a React Native redux implementation in the past, that wouldn't happen, so it wouldn't be necessary to check against the panel type each time the `TabDisplayView` `newState()` method fires. 🤔 

It's just a bit confusing to work with. I expected the `TabDisplayView` to be able to simply "listen" to the one specific `TabsPanelState` that it cares about, rather than all of them simultaneously. 🤔 

e.g. Logs for removing a tab in the normal panel. Always fires in pairs:
```
** new state for private panel
** new state for normal panel
** new state for private panel
** new state for normal panel
** new state for private panel
** new state for normal panel
...
```

Reply:
newState currently fires for all observers regardless of if they are the same type or not. Most redux implementations have a diffing mechanism so it only dispatches updates when a relevant change in state has occurred, but we don't currently have that. We can add it, and we may have to at some point.